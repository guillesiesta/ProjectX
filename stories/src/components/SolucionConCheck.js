import React, {Component} from 'react';

export default class SolucionConCheck extends Component {

  constructor(props){
    super(props);
    this.state={
      puntuacion:this.props.puntuacion,
      puntuacion_previa:this.props.puntuacion,
    };

        this.handlePuntuacionChange = this.handlePuntuacionChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
  }

  /*componentDidMount(){
    console.log("titulo de la historia: "+this.props.titulo)
  }*/

handlePuntuacionChange(event){
  this.setState({puntuacion:event.target.value});
  //var puntuacion = event.target.value;
  //console.log("puntuacion value:"+this.state.puntuacion)
}

handleSubmit(event) {
    //console.log('Tu puntuacion seleccionada es: '+ this.state.puntuacion);
    //console.log('Tu solucion seleccionada es: '+ this.props.solucion);
    //alert("Puntuación enviada correctamente")
      fetch('https://projectx-wvueafqhpp.now.sh/cambiar_puntuacion', { //cargamos las soluciones del acertijo
            method: 'POST', // or 'PUT'
            body: JSON.stringify({solucion:this.props.solucion,
                                  puntuacion:this.state.puntuacion,
                                  titulo:this.props.titulo}), // data can be `string` or {object}!
            headers:{
              'Content-Type': 'application/json'
            }})
        .then(response => response.json())
        .then(data => {
          this.setState({puntuacion_previa:this.state.puntuacion});
          console.log(data);
          //console.log("La storie es: "+data[0].storie)
          //this.setState({storie:data[0].storie});
        })
        .catch(error => console.error(error))

      event.preventDefault();

  }

  render(){
      return(
          <div className="box-header with-border">
            <div className="row">

            <form>
              <li className="list-group-item">Solución: <strong>{this.props.solucion}</strong> y Puntuacion ACTUAL: <strong>{this.state.puntuacion_previa} %</strong>
                <select className="form-control" id="sel1" onChange={this.handlePuntuacionChange}>
                  <option value="-">-</option>
                  <option value="25">25%</option>
                  <option value="50">50%</option>
                  <option value="75">75%</option>
                  <option value="100">100%</option>
                </select>
              </li>
              <input type="button" onClick={this.handleSubmit} value="Cambiar puntuacion" />
            </form>
            </div>

          </div>
      )
  }
}
