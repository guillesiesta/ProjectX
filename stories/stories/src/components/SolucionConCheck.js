import React, {Component} from 'react';

export default class SolucionConCheck extends Component {

  constructor(props){
    super(props);
    this.state={
      puntuacion:this.props.puntuacion,
      puntuacion_previa:this.props.puntuacion,
    };

        this.handlePuntuacionChange = this.handlePuntuacionChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
  }

handlePuntuacionChange(event){
  this.setState({puntuacion:event.target.value});
  //var puntuacion = event.target.value;
  //console.log("puntuacion value:"+this.state.puntuacion)
}

handleSubmit(event) {
    //console.log('Tu puntuacion seleccionada es: '+ this.state.puntuacion);
    //console.log('Tu solucion seleccionada es: '+ this.props.solucion);

    fetch('http://localhost:5000/cambiar_puntuacion', { //cargamos las soluciones del acertijo
          method: 'POST', // or 'PUT'
          body: JSON.stringify({solucion:this.props.solucion,
                                puntuacion:this.state.puntuacion}), // data can be `string` or {object}!
          headers:{
            'Content-Type': 'application/json'
          }})
      .then(response => response.json())
      .then(data => {
        this.setState({puntuacion_previa:this.state.puntuacion});
        //console.log(data);
        //console.log("La storie es: "+data[0].storie)
        //this.setState({storie:data[0].storie});
      })
      .catch(error => console.error(error))

    event.preventDefault();
  }

  render(){
      return(
          <div className="box-header with-border">
            <div className="row">

            <form>
              <li className="list-group-item">Soluci√≥n: <strong>{this.props.solucion}</strong> y Puntuacion ACTUAL: <strong>{this.state.puntuacion_previa}</strong>
                <select className="form-control" id="sel1" onChange={this.handlePuntuacionChange}>
                  <option value="1">1</option>
                  <option value="2">2</option>
                  <option value="3">3</option>
                  <option value="4">4</option>
                </select>
              </li>
              <input type="button" onClick={this.handleSubmit} value="Cambiar puntuacion" />
            </form>
            </div>

          </div>
      )
  }
}
