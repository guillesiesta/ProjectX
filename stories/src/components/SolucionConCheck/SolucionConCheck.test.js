import React, {Component} from 'react';
import { shallow } from 'enzyme';

import SolucionConCheck from './SolucionConCheck';

import "isomorphic-fetch"

describe('SolucionConCheck', ()=> {
  test('Rendering the simple component', ()=>{

    const wrapper = shallow(
      <SolucionConCheck />
    );

    expect(wrapper).toMatchSnapshot();
  })
})
