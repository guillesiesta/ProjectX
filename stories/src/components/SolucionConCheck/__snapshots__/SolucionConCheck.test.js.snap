// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolucionConCheck Rendering the simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SolucionConCheck />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <form>
          <li
            className="list-group-item"
          >
            Solución: 
            <strong />
             y Puntuacion ACTUAL: 
            <strong>
               %
            </strong>
            <select
              className="form-control"
              id="sel1"
              onChange={[Function]}
            >
              <option
                value="-"
              >
                -
              </option>
              <option
                value="25"
              >
                25%
              </option>
              <option
                value="50"
              >
                50%
              </option>
              <option
                value="75"
              >
                75%
              </option>
              <option
                value="100"
              >
                100%
              </option>
            </select>
          </li>
          <input
            onClick={[Function]}
            type="button"
            value="Cambiar puntuacion"
          />
        </form>
      </div>,
      "className": "box-header with-border",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <li
            className="list-group-item"
          >
            Solución: 
            <strong />
             y Puntuacion ACTUAL: 
            <strong>
               %
            </strong>
            <select
              className="form-control"
              id="sel1"
              onChange={[Function]}
            >
              <option
                value="-"
              >
                -
              </option>
              <option
                value="25"
              >
                25%
              </option>
              <option
                value="50"
              >
                50%
              </option>
              <option
                value="75"
              >
                75%
              </option>
              <option
                value="100"
              >
                100%
              </option>
            </select>
          </li>
          <input
            onClick={[Function]}
            type="button"
            value="Cambiar puntuacion"
          />
        </form>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="list-group-item"
            >
              Solución: 
              <strong />
               y Puntuacion ACTUAL: 
              <strong>
                 %
              </strong>
              <select
                className="form-control"
                id="sel1"
                onChange={[Function]}
              >
                <option
                  value="-"
                >
                  -
                </option>
                <option
                  value="25"
                >
                  25%
                </option>
                <option
                  value="50"
                >
                  50%
                </option>
                <option
                  value="75"
                >
                  75%
                </option>
                <option
                  value="100"
                >
                  100%
                </option>
              </select>
            </li>,
            <input
              onClick={[Function]}
              type="button"
              value="Cambiar puntuacion"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Solución: ",
                <strong />,
                " y Puntuacion ACTUAL: ",
                <strong>
                   %
                </strong>,
                <select
                  className="form-control"
                  id="sel1"
                  onChange={[Function]}
                >
                  <option
                    value="-"
                  >
                    -
                  </option>
                  <option
                    value="25"
                  >
                    25%
                  </option>
                  <option
                    value="50"
                  >
                    50%
                  </option>
                  <option
                    value="75"
                  >
                    75%
                  </option>
                  <option
                    value="100"
                  >
                    100%
                  </option>
                </select>,
              ],
              "className": "list-group-item",
            },
            "ref": null,
            "rendered": Array [
              "Solución: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "strong",
              },
              " y Puntuacion ACTUAL: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " %",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " %",
                ],
                "type": "strong",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="-"
                    >
                      -
                    </option>,
                    <option
                      value="25"
                    >
                      25%
                    </option>,
                    <option
                      value="50"
                    >
                      50%
                    </option>,
                    <option
                      value="75"
                    >
                      75%
                    </option>,
                    <option
                      value="100"
                    >
                      100%
                    </option>,
                  ],
                  "className": "form-control",
                  "id": "sel1",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "-",
                      "value": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "25%",
                      "value": "25",
                    },
                    "ref": null,
                    "rendered": "25%",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "50%",
                      "value": "50",
                    },
                    "ref": null,
                    "rendered": "50%",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "75%",
                      "value": "75",
                    },
                    "ref": null,
                    "rendered": "75%",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "100%",
                      "value": "100",
                    },
                    "ref": null,
                    "rendered": "100%",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "type": "button",
              "value": "Cambiar puntuacion",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <form>
            <li
              className="list-group-item"
            >
              Solución: 
              <strong />
               y Puntuacion ACTUAL: 
              <strong>
                 %
              </strong>
              <select
                className="form-control"
                id="sel1"
                onChange={[Function]}
              >
                <option
                  value="-"
                >
                  -
                </option>
                <option
                  value="25"
                >
                  25%
                </option>
                <option
                  value="50"
                >
                  50%
                </option>
                <option
                  value="75"
                >
                  75%
                </option>
                <option
                  value="100"
                >
                  100%
                </option>
              </select>
            </li>
            <input
              onClick={[Function]}
              type="button"
              value="Cambiar puntuacion"
            />
          </form>
        </div>,
        "className": "box-header with-border",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form>
            <li
              className="list-group-item"
            >
              Solución: 
              <strong />
               y Puntuacion ACTUAL: 
              <strong>
                 %
              </strong>
              <select
                className="form-control"
                id="sel1"
                onChange={[Function]}
              >
                <option
                  value="-"
                >
                  -
                </option>
                <option
                  value="25"
                >
                  25%
                </option>
                <option
                  value="50"
                >
                  50%
                </option>
                <option
                  value="75"
                >
                  75%
                </option>
                <option
                  value="100"
                >
                  100%
                </option>
              </select>
            </li>
            <input
              onClick={[Function]}
              type="button"
              value="Cambiar puntuacion"
            />
          </form>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="list-group-item"
              >
                Solución: 
                <strong />
                 y Puntuacion ACTUAL: 
                <strong>
                   %
                </strong>
                <select
                  className="form-control"
                  id="sel1"
                  onChange={[Function]}
                >
                  <option
                    value="-"
                  >
                    -
                  </option>
                  <option
                    value="25"
                  >
                    25%
                  </option>
                  <option
                    value="50"
                  >
                    50%
                  </option>
                  <option
                    value="75"
                  >
                    75%
                  </option>
                  <option
                    value="100"
                  >
                    100%
                  </option>
                </select>
              </li>,
              <input
                onClick={[Function]}
                type="button"
                value="Cambiar puntuacion"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Solución: ",
                  <strong />,
                  " y Puntuacion ACTUAL: ",
                  <strong>
                     %
                  </strong>,
                  <select
                    className="form-control"
                    id="sel1"
                    onChange={[Function]}
                  >
                    <option
                      value="-"
                    >
                      -
                    </option>
                    <option
                      value="25"
                    >
                      25%
                    </option>
                    <option
                      value="50"
                    >
                      50%
                    </option>
                    <option
                      value="75"
                    >
                      75%
                    </option>
                    <option
                      value="100"
                    >
                      100%
                    </option>
                  </select>,
                ],
                "className": "list-group-item",
              },
              "ref": null,
              "rendered": Array [
                "Solución: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "strong",
                },
                " y Puntuacion ACTUAL: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " %",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " %",
                  ],
                  "type": "strong",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value="-"
                      >
                        -
                      </option>,
                      <option
                        value="25"
                      >
                        25%
                      </option>,
                      <option
                        value="50"
                      >
                        50%
                      </option>,
                      <option
                        value="75"
                      >
                        75%
                      </option>,
                      <option
                        value="100"
                      >
                        100%
                      </option>,
                    ],
                    "className": "form-control",
                    "id": "sel1",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "-",
                        "value": "-",
                      },
                      "ref": null,
                      "rendered": "-",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "25%",
                        "value": "25",
                      },
                      "ref": null,
                      "rendered": "25%",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "50%",
                        "value": "50",
                      },
                      "ref": null,
                      "rendered": "50%",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "75%",
                        "value": "75",
                      },
                      "ref": null,
                      "rendered": "75%",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "100%",
                        "value": "100",
                      },
                      "ref": null,
                      "rendered": "100%",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onClick": [Function],
                "type": "button",
                "value": "Cambiar puntuacion",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
